number, string: add S.min and S.max

davidchambers opened the pull request:
```haskell
min :: Ord a => a -> a -> a
max :: Ord a => a -> a -> a
```

[`R.min`][1] and [`R.max`][2], but with type checking. I want to tick some boxes in #145. :)


[1]: http://ramdajs.com/docs/#min
[2]: http://ramdajs.com/docs/#max

davidchambers commented on index.js:
```diff
@@ -169,6 +169,16 @@
+             type === 'String';
+    }
+  );
+
```
We should document all the type classes at some point.

svozza replied:
Agreed. This will also be a very useful type.

svozza commented on test/index.js:
```diff
@@ -3575,6 +3575,80 @@ describe('number', function() {
+
+    it('may be applied to strings', function() {
+      eq(S.min('abc', 'xyz'), 'abc');
+      eq(S.min('xyz', 'abc'), 'abc');
```
Maybe add a case sensitive test?

davidchambers replied:
:zap:

davidchambers commented:
Prompted by your comment, @svozza, I added several more assertions. I also updated the examples to highlight the difference between number comparison and string comparison, and added a paragraph to each function's description to explain the distinction.

svozza commented:
LGTM.
